packages/rum-core/src/domain/vital/vitalCollection.ts

```diff
@@ -9,61 +9,91 @@ import { PageState } from '../contexts/pageStateHistory'
 
 export interface DurationVitalStart {
   name: string
-  startClocks: ClocksState
   context?: Context
+  details?: string
 }
 
 export interface DurationVitalStop {
+  context?: Context
+  details?: string
+}
+
+export interface DurationVitalAdd {
   name: string
-  stopClocks: ClocksState
+  startClocks: ClocksState
+  duration: Duration
   context?: Context
+  details?: string
+}
```

Comment:
**💬 suggestion:** ‏to avoid an intermediary data structure, I would just use `DurationVital` in place of `DurationVitalAdd`. This implies adding the `type` in `rumPublicApi`, but I think it's alright?